packages/gasket-plugin-webpack/lib/index.d.ts

```diff
@@ -1,21 +1,73 @@
 import type { GasketConfig, HookExecTypes } from '@gasket/engine';
+import { Gasket } from '@gasket/engine';
 import type WebpackApi from 'webpack';
+import type { Configuration, Compiler } from 'webpack';
+import type WebpackChain from 'webpack-chain';
 
 export interface WebpackContext {
-  webpack: typeof WebpackApi,
+  webpack: typeof WebpackApi;
   /** @deprecated use require('webpack-merge') */
-  webpackMerge: any
+  webpackMerge: any;
+  isServer?: boolean;
 }
 
 declare module '@gasket/engine' {
   export interface GasketConfig {
-    webpack?: any
+    webpack?: any;
   }
 
   export interface HookExecTypes {
     webpackConfig(
-      config: WebpackApi.Configuration,
+      config: Configuration,
       context: WebpackContext
-    ): WebpackApi.Configuration;
+    ): Configuration;
+    webpackChain(chain: WebpackChain, context: object): void;
+    webpack(baseConfig: object, context: object): void;
+    metrics(metrics: WebpackMetrics): Promise<void>;
   }
 }
+
+declare module '@gasket/plugin-webpack' {
+  /** Creates the webpack config */
+  export function initWebpack(
+    gasket: Gasket,
+    /** Initial webpack config */
+    initConfig: Configuration,
+    /** Additional context-specific information */
+    context: any
+  ): Configuration;
+}
+
+/** Sets up a context object with special getters */
+export function setupContext(
+  gasket: Gasket,
+  /** Additional context-specific information */
+  context: any,
+  name: string
+): any;
+
+export function deprecatedMerges(
```

Comment:
Seems likes this and others are internal private functions. Their types should not be exposed as if they are available. Move to an internal-only types file.